core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -558,7 +558,9 @@ object Battle {
         }
 
         // Remove improvements, add fallout
-        tile.improvement = null
+        if (tile.improvement != null && !attacker.getCivInfo().gameInfo.ruleSet.tileImprovements[tile.improvement]!!.hasUnique("Indestructible")) {
```

Comment:
Here too, same as below, !attacker.getTile().getTileImprovement()!!.hasUnique("Indestructible")

Reply:
I probably should have thought of that, should be fixed now.